### PR TITLE
Fixed subtle Eigen bug

### LINKED ISSUE #261: Memory corruption?
In PR #224, a test in testSpiderWalking gave inconsistent results. I spent quite some tracking it down to a subtle bug in initContactPointGoal, PR with fix coming.

### PR DESCRIPTION
**Pull request by dellaert**
Fixed bug where auto did the wrong thing, and contact goal points were uninitialized
Closes #261

### CODE REVIEW

File: gtdynamics/utils/WalkCycle.cpp [hunk 1]
```diff
@@ -44,8 +45,7 @@ std::map<string, Point3> WalkCycle::initContactPointGoal(
       // If no goal set yet, add it here
       if (cp_goals.count(link_name) == 0) {
         LinkSharedPtr link = robot.link(link_name);
-        auto foot_w = link->bMcom() * kv.second.point +
-                      Point3(0, 0, -ground_height);
+        const Point3 foot_w = link->bMcom() * kv.second.point + adjust;
```
Comment (dellaert) [inline gtdynamics/utils/WalkCycle.cpp:48]: auto probably created an Eigen Expression that did the wrong thing in combination with emplace below. Very evil!

File: gtdynamics/utils/WalkCycle.cpp [hunk 2]
```diff
@@ -36,6 +36,7 @@ void WalkCycle::print(const string &s) const {
 std::map<string, Point3> WalkCycle::initContactPointGoal(
     const Robot &robot, double ground_height) const {
   std::map<string, Point3> cp_goals;
+  const Point3 adjust(0, 0, -ground_height);
```
Comment (varunagrawal) [inline gtdynamics/utils/WalkCycle.cpp:39]: Add a comment about what this is for?

File: tests/testWalkCycle.cpp [hunk 3]
```diff
@@ -52,19 +52,25 @@ TEST(WalkCycle, inverse_kinematics) {
...
-  Point3 goal_RF(0.6, -0.15, 0);  // RF
-  Point3 goal_RH(0, -0.15, 0);    // RH
+  // Expected contact goal points.
+  Point3 goal_LH(-0.371306, 0.1575, 0);   // LH
+  Point3 goal_LF(0.278694, 0.1575, 0);    // LF
+  Point3 goal_RF(0.278694, -0.1575, 0);   // RF
+  Point3 goal_RH(-0.371306, -0.1575, 0);  // RH
+
+  // Check initalization of contact goals.
+  auto cp_goals = walk_cycle.initContactPointGoal(robot, -0.191839);
```
Comment (varunagrawal) [inline tests/testWalkCycle.cpp:62]: Random value of `-0.191839`?

**Review by varunagrawal — APPROVED** [review]
That's interesting. I'm curious about how you found this, but except for some clarifying documentation, it LGTM.

File: gtdynamics/utils/WalkCycle.cpp (hunk 2 above)
Comment (dellaert) [reply gtdynamics/utils/WalkCycle.cpp:39]: Commented in follow-up PR

File: tests/testWalkCycle.cpp (hunk 3 above)
Comment (dellaert) [reply tests/testWalkCycle.cpp:62]: No, height of feet in rest config from SDF

**Comment by dellaert** [comment]
> That's interesting. I'm curious about how you found this, but except for some clarifying documentation, it LGTM.

Lot's of print-debugging. Also, see #264

**Pull request merged**